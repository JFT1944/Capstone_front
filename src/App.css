.App {
  text-align: center;
  background: #fffff4;
  color: #0F683A;
}

.App-logo {
  height: 40vmin;
  pointer-events: none;
}

@media (prefers-reduced-motion: no-preference) {
  .App-logo {
    animation: App-logo-spin infinite 20s linear;
  }
}

.App-header {
  background-color: #282c34;
  min-height: 100vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
  color: white;
}

.App-link {
  color: #61dafb;
}

@keyframes App-logo-spin {
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(360deg);
  }
}

.dispensa_holder{
  width: 100vw;
  min-height:100vh;
  display: flex;
  justify-content: center;
  align-items: center;
}

.NavBar_holder{
  height: 100px;
  width: 100vw;
  display: flex;
  justify-content: flex-end;
  background: rgba(255, 255, 244, .97);
  align-items: center;
  position: fixed;
  top: 0;
  right: 10px;
  z-index: 21;
}
.NavBar_holde-m{
  height: 100vh;
  width: 100vw;
  display: flex;
  justify-content: space-evenly;
  flex-direction: column;
  align-items: center;
  position: fixed;
  top: 10px;
  right: 10px;
  z-index: 21;
}
.navs{
  margin: 0 15px;
  text-decoration: none;
  color:#0F683A;
  font-weight: bold;
}
.logo{
  width: 300px;
  height: 100px;
  /* background: red; */
  position: fixed;
  top: 0px;
  left: 10px;
  display: flex;
  justify-content: flex-start;
  align-items: center;
  z-index: 22;
}
/* .logoM{
  width: 100vw;
  height: 100vh;
  background: red;
  position: fixed;
  top: 0;
  left: 0;
  display: flex;
  justify-content: center;
  align-items: center;
  z-index: 10;
} */

.logo1{
  font-family: Lato;
  font-size: 45px;
  color: #0F683A;
}
.logo2{
color: #AE2C35;
}
.ingredient_card{
  width: 250px;
  border: 3px solid #0F683A;
  border-radius: 25px;
  margin: 12.5px;
  padding: 12.5px;
  box-shadow: 2px 2px 2px rgba(0, 0, 0, .5), -2px -2px 2px rgba(0, 0, 0, .5);
  /* box-shadow: ; */
}
.ingredient_card_holder{
  width: 100vw;
  display: flex;
  flex-wrap: wrap;
  justify-content:center;
  /* padding: 25px; */
  position: absolute;
  top: 130px;
  background: #fffff4;
  
}



.iForms{
  display: flex;
  flex-direction: column;
}
.iForms > span {
  display: flex;
  /* flex-wrap: wrap; */
  flex-direction: column;
  justify-content: space-between;
  align-items: flex-start;
  font-weight: bold;
margin: 10px 0;
}
.iForms > span > input{
  width: 40vw;
  min-width: 275px;
  height: 30px;
  margin-top: 5px;

}
.iForms > span > textarea{
  width: 40vw;
  min-width: 275px;
  height: 350px;
  margin-top: 5px;

}
.iForms > span > select{
  width: 40vw;
  min-width: 275px;
  height: 30px;
  margin-top: 5px;

}
.submit_button > input{
  border: 2px solid #0F683A;
  color: #0F683A;
  font-weight: bold;
  background: none;
}
.submit_button > button{
  border: 2px solid #0F683A;
  color: #0F683A;
  font-weight: bold;
  background: none;
}
.submit_button > input:hover{
background-color: #0F683A;
color: beige;
}

.updatePanelHolder{
  display: flex;
  width: 100vw;
  height: 100vh;
  justify-content: center;
  align-items: center;
  position: fixed;
  z-index: 20;

}
/* .updatePanel{
  background: white;
  border: #0F683A solid 3px;
  width: 300px;
  height: 80vh;
  min-height: 400px;
   */
/* } */
.mobile_button{
/* display: none; */
position: fixed;
top: 25px;
right: 20px;
z-index: 70;
display: none;
}
.mobile_button > img {
  height: 75px;
}
.active{
  transform: scale(1.1);
  color: #AE2C35;
}
#signin_holder > a{
  display: none;

}

.r_add_button{
background: none;
color: #0F683A;
border: #0F683A solid 2px;
font-weight: bold;
}
.r_add_button:hover{
  background-color: #0F683A;
  color: beige;

}
.r_remove_button{
  margin-top: 10px;
  background: none;
  color:#AE2C35;
  font-weight: bold;
  border:#AE2C35 solid 2px;
}
.r_remove_button:hover{
  background-color: #AE2C35;
  color: beige;

}










/* *********** */
/* Media Query Below */



@media screen and (max-width: 1060px){
  .logoM{
    display: none;
  }
  .NavBar_holder{
    display: none;
  }
  .mobile_button{
    display: block;
  }
  #signin_holder > a{
    display: block;
  
  }
  
}

