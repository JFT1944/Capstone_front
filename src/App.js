
import { useEffect, useState } from 'react';
import './App.css';
import DispensaHolder from './Dispensa_Holder';
import { BrowserRouter } from 'react-router-dom';

function App() {
  // hold user information
const [user, setUser] = useState()
// hold token information
const [token, setToken] = useState()

useEffect(() =>{
let lStorage = localStorage.getItem('user')
let tStorage = localStorage.getItem('token')
console.log(tStorage)
// console.log(JSON.parse(lStorage))
if(lStorage){
  setUser(JSON.parse(lStorage))
}
if(tStorage){
  setToken(tStorage)
}

}, [])

  return (
    <div className="App">
      <BrowserRouter>
      <DispensaHolder user={user} setUser={setUser} token={token} setToken={setToken}/>
      </BrowserRouter>
     
    </div>
  );
}







export default App;
