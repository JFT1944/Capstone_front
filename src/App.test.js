// import for tests

import React from 'react';
import { render, fireEvent, getByTestId, screen } from '@testing-library/react';
import App from './App';
import { MemoryRouter } from 'react-router-dom';
process.env.NODE_ENV = 'test';




it('should render without crashing', () => {
    render(<App />)
    
})

it('should match snapshot', () => {
    const {asFragment} = render(<App />)
    expect(asFragment()).toMatchSnapshot()
})
// ############################################
// this test must increments the signup username by 1 each time to work
// ############################################

// it('should render the signup page', async () => {
//     render(<App />)
//     expect(screen.getByText('Sign Up')).toBeInTheDocument()
//     let signup = await screen.findByText('Sign Up')
//     fireEvent.click(signup)
//     expect(await screen.findByText('Sign Up!')).toBeInTheDocument()
//     let b1 = await screen.findByTestId('su1')
//     let b2 = await screen.findByTestId('su2')
//     let b3 = await screen.findByTestId('su3')
//     let b4 = await screen.findByTestId('su4')
//     let b5 = await screen.findByTestId('su5')
//     let b6 = await screen.findByTestId('su6')
//     // must increment the username by 1 each time to work
//     fireEvent.change(b1, {target: {value: 'jp1006'}})
//     fireEvent.change(b2, {target: {value: 'jpjpjp'}})
//     fireEvent.change(b3, {target: {value: 'Justin'}})
//     fireEvent.change(b4, {target: {value: 'Peters'}})
//     fireEvent.change(b5, {target: {value: 'jp@gmail.com'}})

//     fireEvent.click(b6)
//     expect(await screen.findByText(`Welcome 
//         Justin`)).toBeInTheDocument()
// })

it('Navbar => from Login to each page.', async () => {
    
    // let setUser = jest.fn()
    // jest.spyOn(React, 'setUser').mockImplementationOnce((user) => [user, setUser])

    render(<App />)
    let Login = await screen.findByText('Login')
    let signup = await screen.findByText('Sign Up')
    expect(Login).toBeInTheDocument()
    expect(signup).toBeInTheDocument()
    fireEvent.click(signup)
    expect(await screen.findByText('Sign Up!')).toBeInTheDocument()
    fireEvent.click(Login)
    expect(await screen.findByText('Login!')).toBeInTheDocument()
    let username_input = await screen.findByPlaceholderText('Username')
    let password_input = await screen.findAllByPlaceholderText('Password')
    fireEvent.change(username_input, {target: {value: 'jp'}})
    fireEvent.change(password_input[0], {target: {value: 'jpjpjp'}})
    console.log({newp:password_input[0], newu:username_input})
    let submit = await screen.findByTestId('subBut')
    fireEvent.click(submit)
    // setUser({username: 'jp', password: 'jpjpjp'})
    expect(await screen.findByText('Welcome Justin')).toBeInTheDocument()
    expect(await screen.findByText('Add Single Ingredient')).toBeInTheDocument()
    let b1 = await screen.findByText('Add Single Ingredient')
    fireEvent.click(b1)
    expect(await screen.findByPlaceholderText('Name of ingredient')).toBeInTheDocument()
    let b2 = await screen.findByText('Add Recipe Ingredients')
    fireEvent.click(b2)
    expect(await screen.findByPlaceholderText('Recipe Name (Not Necessary)')).toBeInTheDocument()
    let b3 = await screen.findByText('Recipes')
    fireEvent.click(b3)
    expect(await screen.findByText('Cherry Listerine')).toBeInTheDocument()
    let b4 = await screen.findByText('Pantry')
    fireEvent.click(b4)
    expect(await screen.findByText('turkey')).toBeInTheDocument()
    let b5 = await screen.findByText('Shopping List')
    fireEvent.click(b5)
    expect(await screen.findByText('clover')).toBeInTheDocument()
    let b6 = await screen.findByText('Log Out')
    fireEvent.click(b6)
    expect(await Login).toBeInTheDocument()
    


    
    // let b2 = await screen.findByText('Add Recipe Ingredients')
    // let b3 = await screen.findByText('Recipes')
    // let b4 = await screen.findByText('Pantry')
    // let b5 = await screen.findByText('Shopping List')
    // fireEvent.click(b1)
    // expect(await screen.findByText('Name')).toBeInTheDocument()
    })