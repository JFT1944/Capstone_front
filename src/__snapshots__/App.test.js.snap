// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="App"
  >
    <div
      class="dispensa_holder"
    >
      <div
        class="logo"
      >
        <span
          class="logo1"
        >
          Dispensa
          <span
            class="logo2"
          >
            .io
          </span>
        </span>
      </div>
      <div
        class="NavBar_holder"
      >
        <a
          aria-current="page"
          class="navs active"
          href="/"
        >
          Home
        </a>
        <a
          class="navs"
          href="/login"
        >
          Login
        </a>
        <a
          class="navs"
          href="/signup"
        >
          Sign Up
        </a>
      </div>
      <div
        id="signin_holder"
        style="width: 100vw; height: 100vh; display: flex; justify-content: center; align-items: center; flex-direction: column; color: rgb(15, 104, 58);"
      >
        <h2>
          Welcome to Dispensa
          <span
            class="logo2"
          >
            .io!
          </span>
        </h2>
        <h3>
          The best place to store your digital pantry and speed up your shopping experience.
        </h3>
        <a
          class=""
          href="/login"
        >
          Login:
        </a>
        <a
          class=""
          href="/signup"
        >
          Sign Up:
        </a>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
